src/include/cpos.h

```diff
@@ -235,6 +240,57 @@ inline namespace _cpo {
 inline constexpr auto data = _data::_fn{};
 }  // namespace _cpo
 
+// ----------------------------------------------------------------------------
+// num_ids CPO
+// ----------------------------------------------------------------------------
+namespace _num_ids {
+void num_ids(auto&) = delete;
+void num_ids(const auto&) = delete;
+
+struct _fn {
+  template <class T>
+    requires(_member_ids<T>)
+  constexpr auto operator()(T&& t) const noexcept {
+    return t.num_ids();
+  }
+
+  template <class T>
+    requires(!_member_ids<T>)
+  constexpr auto operator()(T&& t) const noexcept {
+    return 0;
+  }
+};
+}  // namespace _num_ids
+inline namespace _cpo {
+inline constexpr auto num_ids = _num_ids::_fn{};
+}  // namespace _cpo
+
+// ----------------------------------------------------------------------------
+// ids CPO
+// @todo Figure out what is wrong with const
+// ----------------------------------------------------------------------------
+namespace _ids {
+void ids(auto&) = delete;
+void ids(const auto&) = delete;
+
+struct _fn {
+  template <class T>
+    requires(_member_ids<T>)
```

Comment:
> If _member_ids checks for num_ids this may not work. 

Sure, updated to use `concept _member_ids = requires(T t) {`.

> And also keeping with convention -- I would put the _member_x concept in the namespace _x

I was following the convention of num_cols and num_rows, are those incorrect? For these short one-liners it seems nicest to just leave as-is, but maybe I'm missing something about why you're suggesting to put it in a namespace.
```
namespace stdx {
using namespace Kokkos;
// using namespace Kokkos::Experimental;
}  // namespace stdx

// ----------------------------------------------------------------------------
// Customization point objects (CPOs) -- implemented as "niebloids"
// ----------------------------------------------------------------------------
template <class T>
concept _member_num_rows = requires(T t) {
  { t.num_rows() } -> semi_integral;
};

template <class T>
concept _member_num_cols = requires(T t) {
  { t.num_cols() } -> semi_integral;
};

template <class T>
concept _member_num_ids = requires(T t) {
  { t.num_ids() };
};

template <class T>
concept _member_ids = requires(T t) {
  { t.ids() };
};
```